lib/double_entry/reporting/aggregate_array.rb

```diff
@@ -50,8 +53,9 @@ def retrieve_aggregates
         scope = LineAggregate.
                 where(:function => function).
                 where(:range_type => 'normal').
-                where(:account => account.to_s).
-                where(:code => code.to_s).
+                where(:account => account.try(:to_s)).
+                where(:code => code.try(:to_s)).
+                where(:partner_account => partner_account.try(:to_s)).
```

Comment:
I think you guys are right, because `partner_account` is an optional filter I think we need to only apply the filter when it's not `nil`, as we are doing here:

https://github.com/envato/double_entry/pull/108/files#diff-449464fb1f62efb0c4b88f08b7295c0dR24

Actually, this is a good point, is there any way we could re-use the `LineAggregateFilter` rather than re-implementing these filters here?